roles/sap_hypervisor_node_preconfigure/tasks/platform/redhat_ocp_virt/install-cnv-operator.yml

```diff
@@ -40,26 +51,62 @@
+- name: Get Install Plan Name from Subscription
   retries: 10
   delay: 10
-  ansible.builtin.command: oc get subscriptions/hco-operatorhub --namespace openshift-cnv --output=jsonpath='{$.status.installplan.name}'
+  kubernetes.core.k8s_info:
+    host: "{{ sap_hypervisor_node_preconfigure_ocp_endpoint }}"
+    ca_cert: "{{ sap_hypervisor_node_preconfigure_ocp_ca_cert }}"
+    api_key: "{{ __sap_hypervisor_node_preconfigure_register_openshift_auth_results.openshift_auth.api_key }}"
+    kubeconfig: "{{ sap_hypervisor_node_preconfigure_ocp_kubeconfig }}"
+    api_version: operators.coreos.com/v1alpha1
+    kind: Subscription
+    name: hco-operatorhub
+    namespace: openshift-cnv
   register: __sap_hypervisor_node_preconfigure_register_cnv_subscription_install_plan_name
-  until: __sap_hypervisor_node_preconfigure_register_cnv_subscription_install_plan_name.stdout != ""
-  changed_when: __sap_hypervisor_node_preconfigure_register_cnv_subscription_install_plan_name.stdout != ""
+  until: __sap_hypervisor_node_preconfigure_register_cnv_subscription_install_plan_name.resources[0].status.installPlanRef.name is defined
+  changed_when: __sap_hypervisor_node_preconfigure_register_cnv_subscription_install_plan_name.resources[0].status.installPlanRef.name is defined
+
+- name: Set Install Plan Name
+  ansible.builtin.set_fact:
+    install_plan_name: "{{ __sap_hypervisor_node_preconfigure_register_cnv_subscription_install_plan_name.resources[0].status.installPlanRef.name }}"
 
 - name: Wait for Install Plan to finish
-  ansible.builtin.command: "oc wait installplan \
-    {{ __sap_hypervisor_node_preconfigure_register_cnv_subscription_install_plan_name.stdout }} --namespace openshift-cnv --for=condition='Installed' --timeout='5m'"
-  register: __sap_hypervisor_node_preconfigure_register_wait_for_installplan
-  changed_when: __sap_hypervisor_node_preconfigure_register_wait_for_installplan.rc != 0
+  vars:
+    install_plan_name: "{{ __sap_hypervisor_node_preconfigure_register_cnv_subscription_install_plan_name.stdout }}"
+    namespace: "openshift-cnv"
+  block:
+    - name: Get Install Plan details
+      kubernetes.core.k8s_info:
+        host: "{{ sap_hypervisor_node_preconfigure_ocp_endpoint }}"
+        ca_cert: "{{ sap_hypervisor_node_preconfigure_ocp_ca_cert }}"
+        api_key: "{{ __sap_hypervisor_node_preconfigure_register_openshift_auth_results.openshift_auth.api_key }}"
+        kubeconfig: "{{ sap_hypervisor_node_preconfigure_ocp_kubeconfig }}"
+        api_version: operators.coreos.com/v1alpha1
+        kind: InstallPlan
+        name: "{{ install_plan_name }}"
+        namespace: "{{ namespace }}"
+      register: __sap_hypervisor_node_preconfigure_register_wait_for_installplan
+      until: __sap_hypervisor_node_preconfigure_register_wait_for_installplan.resources[0].status.phase == "Complete"
+      retries: 60
+      delay: 5
+      ignore_errors: true
+
+    - name: Fail if Install Plan is not Complete after waiting
```

Comment:
Could this be avoided by adding a `failed_when` above?

Reply:
failed_when is not a generic attribute, moreover, it's not available for kubernetes.core.k8s_info